docs/sso_via_saml_mscolab.rst

```diff
@@ -0,0 +1,557 @@
+         name_id_format_allow_create: true
+
+.. note::
+    Make sure to update 
+    entityid : 'idp_identity_name' 
+    Assertion_consumer_service : with the urls of assertion consumer services functionalities URL that going to implement next step, may be better to explain here
+
+    Key_file : if need can be update through the server
+    Cert_file : if need can be update through the server 
+    Metadata.local : if need can be update through the server
+
+
+Access SAML2Client metadata of MSColab
+--------------------------------------
+
+While the core purpose of IdPs is to authenticate users and provide information to relying parties, the responses can vary based on configuration, protocol, user attributes, consent, and customization. Therefore, responses from different IdPs can indeed be different, and developers and administrators should be aware of these variations when integrating with different identity providers. However, in the MSColab server, we implemented an easy way to access metadata from an endpoint. You can access it easily by using the specified url, which is configured based on the settings of your SAML2 client in your `setupsaml2backend.py` and `saml2backend.yaml` file. This streamlined approach simplifies the process and eliminates the need for manual development of endpoints and functionalities specific to each IdP.
+
+.. note::
+    URL to access metadata endpoint for particular IdP:
+    ``/metadata/<idp_identity_name>``
+
+Guide to IDP Configuration
+--------------------------
+
+In the SSO process through the MSColab server, the username is obtained as ``givenName``, and the email address is obtained as ``email``. Therefore, when configuring the IdP, it is necessary to configure it accordingly to ensure the correct return of the givenName attribute and the email address along with the SAML response.
+
+
+3. Configuration example through Keycloak 13.0.1
+************************************************
+
+Setting Up Keycloak
+-------------------
+
+Installation and run Keycloak
+#############################
+
+Via local installation
+    1. Download the file (requires java, wget installed):
+
+    .. code:: text
+
+        cd $HOME && \ wget -c keycloak_13_0_1.tar.gz https://github.com/keycloak/keycloak/releases/download/13.0.1/keycloak-13.0.1.tar.gz -O - | tar -xz
+
+|
+
+    2. Navigate to the KeyCloak binaries folder:
+
+    .. code:: text
+
+        cd keycloak-13.0.1/bin
+
+|
+
+    3. And start it up:
+
+    .. code:: text
+
+        ./standalone.sh
+
+|
+
+Via Docker (requires Docker installed)
+    * Open your terminal and run
+
+    .. code:: text
+
+        docker run -p 8080:8080 -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin quay.io/keycloak/keycloak:13.0.1
+
+|
+
+    .. image:: images/sso_via_saml_conf/ss_docker_run_cmd.png
+        :width: 400
+
+.. note::
+    You can define KEYCLOAK_USER and KEYCLOAK_PASSWORD as you wish.
```

Comment:
maybe add this before the docker call, admin:admin is high risk, e.g. use pwgen, makepasswd

greenlight uses it that way so it can be configured from outside https://github.com/bigbluebutton/greenlight/blob/master/keycloak.nginx

unfortunately people doing that will forget about that simple password